secure/secure_filter_test.go

```diff
@@ -0,0 +1,92 @@
+package secure
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/google/martian"
+	"github.com/google/martian/martiantest"
+	"github.com/google/martian/session"
+)
+
+func TestSecureFilter(t *testing.T) {
+	req, err := http.NewRequest("GET", "http://martian.local", nil)
+	if err != nil {
+		t.Fatalf("http.NewRequest(): got %v, want no error", err)
+	}
+
+	ctx, err := session.FromContext(nil)
+	if err != nil {
+		t.Fatalf("session.FromContext(): got %v, want no error", err)
+	}
+
+	martian.SetContext(req, ctx)
+
+	session := ctx.GetSession()
+
+	f := NewSecureFilter()
+	tm := martiantest.NewModifier()
+	f.SetRequestModifier(tm)
+
+	if err := f.ModifyRequest(req); err != nil {
+		t.Fatalf("ModifyRequest(): got %v, want no error", err)
+	}
+
+	if got := tm.RequestModified(); got == true {
```

Comment:
it tm.RequestModified() {
  t.Error("tm.RequestModified(): got true, want false")
}